From 0940ea1e54e209d8e8fa97e2ebaa8bf98da62e34 Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Tue, 30 Dec 2025 22:06:18 +0800
Subject: [PATCH 50/63] loonggpu: don't return an error if backlight
 initialization fails

If this returns an error, the load of loonggpu will fail.  But in fact
it should still work without backlight support.

To make things worse, the error handling logic of loonggpu seems buggy
as at now (triggering many WARNINGs and Oops).  It should be fixed in
the future.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 loonggpu/loonggpu_backlight.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/loonggpu/loonggpu_backlight.c b/loonggpu/loonggpu_backlight.c
index e54029a..e24421a 100644
--- a/loonggpu/loonggpu_backlight.c
+++ b/loonggpu/loonggpu_backlight.c
@@ -487,17 +487,16 @@ int loonggpu_backlight_register(struct drm_connector *connector)
 	struct loonggpu_backlight *ls_bl;
 	struct loonggpu_device *adev = connector->dev->dev_private;
 	struct backlight_resource *bl_res;
-	int ret = 0;
 	bool ls_bl_status = false;
 
 	bl_res = dc_get_vbios_resource(adev->dc->vbios, connector->index,
 			LOONGGPU_RESOURCE_BACKLIGHT);
 	if (!bl_res)
-		return ret;
+		return 0;
 
 	ls_bl = loonggpu_backlight_init(adev, connector->index);
 	if (!ls_bl)
-		return -ENXIO;
+		return 0;
 
 	ls_bl_status = loonggpu_backlight_get_hw_status(ls_bl);
 	if (ls_bl_status) {
@@ -512,5 +511,5 @@ int loonggpu_backlight_register(struct drm_connector *connector)
 		 ls_bl_status ? "on" : "off",
 		 ls_bl->hw_enabled ? "on" : "off");
 
-	return ret;
+	return 0;
 }
-- 
2.52.0

